e2e_playwright/st_plotly_chart_select.py

```diff
@@ -0,0 +1,189 @@
+np.random.seed(0)
+
+df_bubble = px.data.gapminder()
+fig_bubble = px.scatter(
+    df_bubble.query("year==2007"),
+    x="gdpPercap",
+    y="lifeExp",
+    size="pop",
+    color="continent",
+    hover_name="country",
+    log_x=True,
+    size_max=60,
+)
+fig_bubble.update_layout(dragmode="select")
+st.header("Bubble Chart with Box Select")
+st.plotly_chart(fig_bubble, on_select="rerun", key="bubble_chart", selection_mode="box")
+if len(st.session_state.bubble_chart.select["points"]) > 0:
+    st.write("The original df data selected:")
+    points = st.session_state.bubble_chart.select["points"]
+    # Extract x and y values directly into lists
+    x_values = [point["x"] for point in points]
+    y_values = [point["y"] for point in points]
+
+    # Use these lists to filter the DataFrame
+    filtered_df = df_bubble[
+        df_bubble["gdpPercap"].isin(x_values) & df_bubble["lifeExp"].isin(y_values)
+    ]
+    st.dataframe(filtered_df)
+else:
+    st.write("Nothing is selected")
+
+st.header("Line Chart with Lasso select")
+df = px.data.gapminder().query("continent=='Oceania'")
+fig_linechart = px.line(df, x="year", y="lifeExp", color="country", markers=True)
+# Update the configuration to enable lasso selection
+fig_linechart.update_layout(dragmode="lasso")
+st.plotly_chart(
+    fig_linechart, on_select="rerun", key="line_chart", selection_mode=["lasso"]
+)
+if len(st.session_state.line_chart.select["points"]) > 0:
+    st.write("The original df data selected:")
+    points = st.session_state.line_chart.select["points"]
+    # Extract x and y values directly into lists
+    x_values = [point["x"] for point in points]
+    y_values = [point["y"] for point in points]
+
+    # Use these lists to filter the DataFrame
+    filtered_df = df[df["year"].isin(x_values) & df["lifeExp"].isin(y_values)]
+    st.dataframe(filtered_df)
+else:
+    st.write("Nothing is selected")
+
+st.header("Bar Chart with Click")
+data_canada = px.data.gapminder().query("country == 'Canada'")
+fig_bar = px.bar(data_canada, x="year", y="pop")
+st.plotly_chart(fig_bar, on_select="rerun", key="bar_chart", selection_mode=["points"])
+if len(st.session_state.bar_chart.select["points"]) > 0:
+    st.write("The original df data selected:")
+    points = st.session_state.bar_chart.select["points"]
+    # Extract x and y values directly into lists
+    x_values = [point["x"] for point in points]
+    y_values = [point["y"] for point in points]
+
+    # Use these lists to filter the DataFrame
+    filtered_df = data_canada[
+        data_canada["year"].isin(x_values) & data_canada["pop"].isin(y_values)
+    ]
+    st.dataframe(filtered_df)
+else:
+    st.write("Nothing is selected")
+
+
+st.header("Box Selections for a Stacked Bar Chart")
+wide_df = px.data.medals_wide()
+fig = px.bar(
+    wide_df, x="nation", y=["gold", "silver", "bronze"], title="Wide-Form Input"
+)
+fig.update_layout(dragmode="select")
+st.plotly_chart(
```

Comment:
All of these tests set the `key` and use the session state value. For better test coverage, it would be good to have one of these tests use the return value instead + not set the key.